pandas/core/arrays/arrow/array.py

```diff
@@ -2915,7 +2915,11 @@ def _dt_to_pytimedelta(self) -> np.ndarray:
         return np.array(data, dtype=object)
 
     def _dt_total_seconds(self) -> Self:
-        result = pa.array(self._to_timedeltaarray().total_seconds(), from_pandas=True)
+        # Convert duration to seconds using PyArrow compute
+        unit = self._pa_array.type.unit
+        unit_to_seconds = {"s": 1.0, "ms": 1e-3, "us": 1e-6, "ns": 1e-9}
+        factor = unit_to_seconds[unit]
+        result = pc.multiply(pc.cast(self._pa_array, pa.int64()), factor)
```

Comment:
Might want to do a division by 1e9 instead of multiply by 1e-9, I think that gives a better floating point precision behaviour